### ISSUE TITLE
"Error: filename must not contain illegal characters" when a filename contain a legal character

### ISSUE THREAD

**Issue by ratuka** [downthemall/downthemall#492]
**Desktop (please complete the following information):**
 - OS: Linux (Fedora, Gnome)
 - Browser and version: Firefox 121.0
 - DownThemAll! version: 4.11.1

**Describe the bug**
DownThemAll! can't download a file which has (for example) a hungarian "ő" (%C5%91) character in the URL.

**Comment by nmaier** [comment]
Do you have specific example urls?

**Comment by ratuka** [comment]
https://canadahun.com/attachments/roberto-ricci-a-v%C3%B6r%C3%B6s-harlekin-1-maszkokr%C3%B3l-%E2%80%8B%C3%A9s-f%C3%B6stenyekr%C5%91l-epub.1986075/

It's a hungarian closed forum, to download you need registration and some post, but the url is problematic with DownThemAll.

**Comment by CuriousG73** [comment]
any title or page that has "&" or ":" characters fails when I use *title* or *pagetitle* mask
example 
Title is TEAM SPIRIT: КОУЧИНГ DOTA 2
https://vk.com/video-32151480_456247403 

Is there a way to convert these to flat in the similar manner as the flat "/" to remove sub-directories ?

**Comment by ratuka** [comment]
https://canadahun.com/attachments/koltay-g%C3%A1bor-john-%E2%80%8Blennon-1940%E2%80%931980-epub.1993442/

**Comment by LucaNonato** [comment]
This also happens on Windows and macOS regardless of the URL: if the destination folder contains one or more "." i get the same error (But this stopped working recently: i had no problem whatsoever creating folders like "Check this out..." using DTA!.

P.S.: this is the right place for my answer or is better to open a separate issue?


Cheers!
Luca

**Comment by LucaNonato** [comment]
>  I've noticed that in older versions of FF this error does not appear no matter what version of the add-on is used

Exactly: can't tell you with which version this started but also occurs on "115.12.0esr", not only on "127.x".

**Comment by DRG80N** [comment]
This is occurring for me on Firefox addon V 4.12.1. which fails all ""Error: filename must not contain illegal characters"

Interestingly, it does not occur for the exact same webpage, the same time on Chrome addon of the same version. - Completes perfectly.

Simplified Chinese characters are on the web page.

**Comment by DRG80N** [comment]
any thoughts on this one? strange that chrome add-in is fine, but firefox is fkd

**Comment by segln** [comment]
Same here on FF 130.

**Comment by LucaNonato** [comment]
I confirm that using Firefox 130.0 (+ Windows 10 22H2) the issue is gone.

**Comment by wave-blessed** [comment]
Firefox 140 ESR. Windows 11. DownThemAll 4.14.2.
If filename contains brackets () the issue is relevant.
